### PR TITLE
Dockerfile: remove custom go build for windows/arm64

### PR DESCRIPTION
**Pull request by tonistiigi**
Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

### CODE REVIEW

**Comment by tonistiigi** [comment]
Weird error

```
#28 2.212 Building static docker-windows-arm64.exe
#28 4.534 # internal/abi
#28 4.534 /usr/local/go/src/internal/abi/abi.go:19:10: undefined: IntArgRegs
#28 4.534 /usr/local/go/src/internal/abi/abi.go:20:10: undefined: FloatArgRegs
#28 4.534 /usr/local/go/src/internal/abi/abi.go:27:8: undefined: IntArgRegs
#28 4.534 /usr/local/go/src/internal/abi/abi.go:38:24: undefined: IntArgRegs
```

Passes if I build on arm64 but fails on amd64.

**Comment by tonistiigi** [comment]
Go issue: https://github.com/golang/go/issues/46815

File: Dockerfile [hunk 1]
```diff
@@ -11,7 +11,7 @@ FROM golatest AS go-darwin
 FROM golatest AS go-windows-amd64
 FROM golatest AS go-windows-386
 FROM golatest AS go-windows-arm
-FROM --platform=$BUILDPLATFORM tonistiigi/golang:497feff1-${BASE_VARIANT} AS go-windows-arm64
+FROM --platform=$BUILDPLATFORM golang:1.17beta1-${BASE_VARIANT} AS go-windows-arm64
```
Comment (thaJeztah) [inline Dockerfile:14]: I see https://github.com/golang/go/issues/46815 was closed; can you try updating to rc1? 

```suggestion
FROM --platform=$BUILDPLATFORM golang:1.17rc1-${BASE_VARIANT} AS go-windows-arm64
```

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
@StefanScherer @silvin-lubecki PTAL

**Review by silvin-lubecki — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
bringing this one in

**Pull request merged**